### ISSUE TITLE
"postfix_smtp_response" parsing issues

### ISSUE THREAD

**Issue by jtryon** [whyscream/postfix-grok-patterns#51]
Seems like there may be issues with this, I started seeing some random terms appearing such as:
*postfix_http://support.google.com/mail/bin/answer.py?hl.raw  
*postfix_http://support.google.com/mail/bin/answer.py?answer.raw  
*postfix_http://postmaster.1and1.com/en/error-messages?ip.raw  
*postfix_http://support.google.com/mail/bin/answer.py?answer  

Here's the full syslog output of the log line I believe generated the first time in my list:
```
2015-04-22T09:44:35.282606-07:00 gallifrey postfix/smtp[62812]: 4E74613FE76: to=<redacted@gmail.com>, relay=gmail-smtp-in.l.google.com[2607:f8b0:400e:c04::1b]:25, delay=1.9, delays=0.05/0/1.1/0.71, dsn=5.7.1, status=bounced (host gmail-smtp-in.l.google.com[2607:f8b0:400e:c04::1b] said: 550-5.7.1 [2607:4100:3:25:100::2      12] Our system has detected that this 550-5.7.1 message is likely unsolicited mail. To reduce the amount of spam sent 550-5.7.1 to Gmail, this message has been blocked. Please visit 550-5.7.1 http://support.google.com/mail/bin/answer.py?hl=en&answer=188131 for 550 5.7.1 more information. qc12si8475009pab.211 - gsmtp (in reply to end of DATA command))
```

The reason I think it's messing with the smtp_response parsing is...  I get these grok'd terms as a result:
*postfix_smtp_response	 	in reply to end of DATA command)
*postfix_http://support.google.com/mail/bin/answer.py?hl	 	en&answer=188131

Hope you have time to take a look.  I'll update this if I figure out a nice fix.

**Comment by whyscream** [comment]
I've seen these too, but didn't have time to actually try and find a solution for them. I need to spend some time on them probably If youhave time and find a solution, please share :)

**Comment by jtryon** [comment]
Okay.. got a little farther.  I'm not too great with grok though. : )
It looks like nested brackets in the smtp response are making it mess up.

(host gmail-smtp-in.l.google.com[2607:f8b0:400e:c04::1b] said: 550-5.7.1 [2607:4100:3:25:100::2      12] Our system has detected that this 550-5.7.1 message is likely unsolicited mail. To reduce the amount of spam sent 550-5.7.1 to Gmail, this message has been blocked. Please visit 550-5.7.1 http://support.google.com/mail/bin/answer.py?hl=en&answer=188131 for 550 5.7.1 more information. qc12si8475009pab.211 - gsmtp (in reply to end of DATA command))

Just need to figure out a way to make sure it matches the FIRST and LAST brackets and not the additional ones nested inside.

**Comment by whyscream** [comment]
I created a possible fix in PR #52, this assumes that the status=(sent|bounced|etc) is alays the last actual key-value pair in the list, optionally followed by the smtp response. Please test (if possible) and report back, I'm trying this in production already

**Comment by whyscream** [comment]
I merged #52 to master today, seems to work without breaking other stuff. Thanks for reporting.

**Comment by jtryon** [comment]
Confirm that it seems to work! : D  Thanks dude.
